### PR TITLE
Refactoring and renaming

### LINKED ISSUE #149: rename some classes
Names like `LM` are confusing

* `LM` -> `Fitter`
* `LMCoellip` -> `CoellipFitter`
* `GMixEM` -> `EMFitter`
* `GMixEMFixCen` -> `EMFitterFixCen`
* `GMixEMFluxOnly` -> `EmFitterFluxOnly`
* `TemplateFluxFitter` -> `PSFFluxFitter` (we can still support non-psf flux fitting)

### PR DESCRIPTION
**Pull request by esheldon**
closes #149 

Renaming some classes.  I also decided to put related modules together into sub-packages.  E.g. we generally keep the files with numba jitted functions in a separate module.

### CODE REVIEW

**Comment by esheldon** [comment]
the mdet test cannot pass until we merge this to master

**Comment by esheldon** [comment]
Now is the time to bikeshed

**Comment by esheldon** [comment]
These are stylistic changes so have at it!

File: ngmix/defaults.py [hunk 1]
```diff
@@ -8,3 +8,20 @@
 # for priors etc.
 LOWVAL = -np.inf
 BIGVAL = 9999.0e47
+
+DEFAULT_LM_PARS = {"maxfev": 4000, "ftol": 1.0e-5, "xtol": 1.0e-5}
+
+ADMOM_RESULT_DTYPE = [
```
Comment (beckermr) [inline ngmix/defaults.py:14]: I'm not 100% sure this default belongs here.

Comment (esheldon) [reply ngmix/defaults.py:14]: This is a result of the refactoring into sub-packages.

I had to take it out of the package to use it in another package due to circular imports.

Comment (esheldon) [reply ngmix/defaults.py:14]: wasn't a problem when they were plain modules, something special about packages

Comment (beckermr) [reply ngmix/defaults.py:14]: ah right. Fine by me ofc. Usually, a circle import indicates a design issue, but I don't think we should care.

Comment (esheldon) [reply ngmix/defaults.py:14]: could make another module `types.py` or something and put others there as well

Comment (esheldon) [reply ngmix/defaults.py:14]: looking again there is  no reason to re-use this; it is I think OK to just copy it

**Review by beckermr — APPROVED** [review]

File: ngmix/defaults.py (hunk 1 above)
Comment (beckermr) [reply ngmix/defaults.py]: Ahhhh! I think that was the design flaw. They are simply not the same thing. Too DRY!

Comment (esheldon) [reply ngmix/defaults.py]: agreed, too dry :)

**Comment by esheldon** [comment]
Its important to turn on the brain when refactoring :)

**Pull request merged**